docker/docker-compose.yml

```diff
@@ -0,0 +1,178 @@
+version: "3.8"
+name: hskauting
+
+networks:
+    traefik:
+
+volumes:
+    mysql:
+    mysql-test:
+
+x-php-base:
+    &php-base
+    build:
+        context: ./containers/php
+    working_dir: /app
+    env_file:
+        - ./containers/php/ini/image.ini
+    networks:
+        - traefik
+    volumes:
+        - ../:/app
+        - ./containers/php/ini/php.ini:/etc/php/8.1/fpm/conf.d/php.ini
+        - ./containers/php/ini/php.ini:/etc/php/8.1/cli/conf.d/php.ini
+
+x-mysql-base:
+    &mysql-base
+    image: mysql:8.0
+    networks:
+        - traefik
+    environment:
+        MYSQL_DATABASE: hskauting
+        MYSQL_USER: hskauting
+        MYSQL_PASSWORD: hskauting
+        MYSQL_ROOT_PASSWORD: root
+        MYSQL_COLLATION: utf8mb3_czech_ci
+        MYSQL_CHARACTER_SET: utf8
+
+services:
+    traefik:
```

Comment:
traefik abychom nemuseli menit etc/hosts?

Reply:
Je to router pro ty kontejnery. Dělá managment vnitřní sítě, zajišťuje vzájemnou viditelnost. Nastavuje *.localhost domény. Hosts změna je vyžádána redirectem zpět ze skautisu a alespoň na WSL nejde tento záznam obejít. Operační systém pak neví, kam požadavek směřovatl. Vevnitř wsl už to funguje. Nevím jak by se to chovalo na čistém linuxu, ale zkusím to zjistit

Reply:
Traefik znám, ale není mi jasné kdo na koho potřebuje vidět na doméně. Protože main container se připojí k db pres název kontejneru a zvenku na main container se pripojí pres /etc/hosts.

Adminer bych hodil jen na jiny port, at nemame kvuli tomu dalsi container co žere pamět

Reply:
Adminer je jediný kontejner, co tam ve výsledku být nemusí. Port ven nemá vůbec. Sestava má pouze dva porty a to 80 pro aplikaci a 3306 pro případné připojení mysql serveru například k IDE

Reply:
Kdyz vystavit adminer na localhost:8080 tak neni potreba pro něj traefik

Reply:
to pak nebudu mít přístupné pod doménou adminer.localhost 
Takhle port neblokuju vůbec

Reply:
abys to mel na domene, tak potrebujes traefik a to bych to radeji nechal jen na vlastnim portu, to nic neblokuje a nebere to prostredky navíc

Reply:
Bez traefiku ta sestava nebude fungovat

Reply:
ale proc?